tensorflow/python/tools/optimize_for_inference_test.py

```diff
@@ -500,6 +501,275 @@ def testFusePadAndConv(self):
+    # folding FusedBatchNorm.
+    optimized_graph_def = optimize_for_inference_lib.fold_batch_norms(
+        optimized_graph_def)
+    for node in optimized_graph_def.node:
+      self.assertNotEqual("FusedBatchNorm", node.op)
+
+    with self.cached_session() as sess:
+      _ = importer.import_graph_def(
+          optimized_graph_def, input_map={}, name="optimized2")
+      optimized_result = sess.run(["optimized2/output:0"])
+
+    self.assertAllClose(original_result, optimized_result,
+                        rtol=1e-04, atol=1e-06)
+
+  @test_util.run_deprecated_v1
+  def testFuseDecomposedBatchNorm_MatchSwitchOrder(self):
+    original_graph_def, original_result = self.create_base_for_fuse_batchnorm(
+        "MATCH_SWITCH_ORDER")
+
+    # Test correctness of fusing individual ops to FusedBatchNorm.
+    optimized_graph_def = optimize_for_inference_lib.fuse_decomposed_batch_norm(
+        original_graph_def)
+
+    batchnorm_count, decompose_count = self.count_batchnorm_relavant_ops(
+        optimized_graph_def)
+    self.assertEqual(batchnorm_count, 1)
+    self.assertEqual(decompose_count, 0)
+
+    with self.cached_session() as sess:
+      _ = importer.import_graph_def(
+          optimized_graph_def, input_map={}, name="optimized")
+      optimized_result = sess.run(["optimized/output:0"])
+
+    self.assertAllClose(original_result, optimized_result)
+
+    # Test correctness of fusing individual ops to FusedBatchNorm followed by
+    # folding FusedBatchNorm.
+    optimized_graph_def = optimize_for_inference_lib.fold_batch_norms(
+        optimized_graph_def)
+    for node in optimized_graph_def.node:
+      self.assertNotEqual("FusedBatchNorm", node.op)
+
+    with self.cached_session() as sess:
+      _ = importer.import_graph_def(
+          optimized_graph_def, input_map={}, name="optimized2")
+      optimized_result = sess.run(["optimized2/output:0"])
+
+    self.assertAllClose(original_result, optimized_result,
+                        rtol=1e-04, atol=1e-06)
+
+  @test_util.run_deprecated_v1
+  def testFuseDecomposedBatchNorm_PatternMismatchCase(self):
+    original_graph_def, original_result = self.create_base_for_fuse_batchnorm(
+        "MISMATCH_PATTERN")
+
+    # Test for not to fuse ops if graph has same types of ops but pattern mismatch.
+    optimized_graph_def = optimize_for_inference_lib.fuse_decomposed_batch_norm(
+        original_graph_def)
+
+    batchnorm_count, math_op_count = self.count_batchnorm_relavant_ops(
+        optimized_graph_def)
+    self.assertEqual(batchnorm_count, 0)
+    self.assertEqual(math_op_count, 7)
+
+    with self.cached_session() as sess:
+      _ = importer.import_graph_def(
+          optimized_graph_def, input_map={}, name="optimized")
+      optimized_result = sess.run(["optimized/output:0"])
+
+    self.assertAllClose(original_result, optimized_result)
+
+  @test_util.run_deprecated_v1
+  def testFuseDecomposedBatchNorm_FormatUnsupportCase(self):
```

Comment:
nit: FormatUnsupportCase -> FormatUnsupportedCase

This test is failing with:
```
tensorflow.python.framework.errors_impl.UnimplementedError: The Conv2D op currently only supports the NHWC tensor format on the CPU. The op was given the format: NCHW
	 [[{{node conv_op}}]]
```

Reply:
The fusion is only implemented for NHWC format, and I added the check for the data format for precaution.
Is it ok not to add the mismatch test case for the data format? 
If it is still needed, do you know any flag that I can add to skip this test case for CPU device?
( I am testing on CPU, couldn't see this test failure).

Reply:
Maybe it's supported with oneDNN, but not without?

Maybe you can condition the test on onednn, or otherwise catch the error and skip if the format itself isn't supported.